inc/az_ulib_ipc_api.h

```diff
@@ -0,0 +1,299 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#ifndef AZ_ULIB_IPC_H
+#define AZ_ULIB_IPC_H
+
+#include "azure_macro_utils/macro_utils.h"
+#include "umock_c/umock_c_prod.h"
+
+#include "az_ulib_action_api.h"
+#include "az_ulib_base.h"
+#include "az_ulib_descriptor_api.h"
+#include "internal/az_ulib_ipc.h"
+#include "ulib_config.h"
+#include "ulib_port.h"
+#include "ulib_result.h"
+
+#ifndef __cplusplus
+#include <stdint.h>
+#else
+#include <cstdint>
+extern "C" {
+#endif /* __cplusplus */
+
+/**
+ * @file    az_ulib_ipc_api.h
+ *
+ * @brief   The inter-process communication.
+ *
+ * The IPC is the component responsible to expose interfaces created by one component to all other
+ * components in the system.
+ */
+
+/**
+ * @brief IPC handle.
+ */
+typedef struct az_ulib_ipc_tag {
+  _az_ulib_ipc az_private;
+} az_ulib_ipc;
+
+/**
+ * @brief Interface handle.
+ */
+typedef _az_ulib_ipc_interface_handle az_ulib_ipc_interface_handle;
+
+/**
+ * @brief   Initialize the IPC system.
+ *
+ * This API initialize the IPC. It shall be called only once, at the beginning of the code
+ * execution.
+ *
+ * @note    This API **is not** thread safe, the other IPC API shall only be called after the
+ * initialization process is completely done.
+ *
+ * @param[in]   ipc_handle      The `az_ulib_ipc *` that points to a memory position where
```

Comment:
>`az_ulib_ipc *` [](start = 36, length = 15)

like the comment about the function pointer, because this is an actual type, you can link to it with a `#`. so in this case you can do `#az_ulib_ipc*` and it will link properly